util/src/main/java/tech/pegasys/artemis/util/config/ArtemisConfiguration.java

```diff
@@ -13,401 +13,338 @@
-        "Metric categories to enable",
-        null);
-
-    // Logging
-    builder.addBoolean(
-        "logging.colorEnabled",
-        true,
-        "Whether Status and Event log messages include a console color display code",
-        PropertyValidator.isPresent());
-    builder.addBoolean(
-        "logging.includeEventsEnabled",
-        true,
-        "Whether the frequent update events are logged (e.g. every slot event, with validators and attestations))",
-        PropertyValidator.isPresent());
-    builder.addString(
-        "logging.destination",
-        "both",
-        "Whether all logs go only to the console, only to the log file, or both",
-        PropertyValidator.anyOf("consoleOnly", "fileOnly", "both"));
-    builder.addString(
-        "logging.file",
-        "teku.log",
-        "Path containing the location (relative or absolute) and the log filename.",
-        PropertyValidator.isPresent());
-    builder.addString(
-        "logging.fileNamePattern",
-        "teku_%d{yyyy-MM-dd}.log",
-        "Pattern for the filename to apply to rolled over logs files.",
-        PropertyValidator.isPresent());
-
-    // Outputs
-    builder.addString(
-        "output.transitionRecordDir",
-        "",
-        "Directory to record transition pre and post states",
-        null);
-
-    // Database
-    builder.addString(
-        "database.dataPath", ".", "Path to output data files", PropertyValidator.isPresent());
-    builder.addString(
-        "database.stateStorageMode",
-        "prune",
-        "Sets the strategy for handling historical chain state.  Supported values include: 'prune', and 'archive'",
-        null);
-
-    // Beacon Rest API
-    builder.addInteger("beaconrestapi.portNumber", 5051, "Port number of Beacon Rest API", null);
-    builder.addBoolean(
-        "beaconrestapi.enableSwagger", false, "Enable swagger-docs and swagger-ui endpoints", null);
-
-    builder.validateConfiguration(
-        config -> {
-          return null;
-        });
-
-    return builder.toSchema();
-  }
-
-  private static final Schema schema = createSchema();
-
-  /**
-   * Reads configuration from file.
-   *
-   * @param path a toml file to read configuration from
-   * @return the new ArtemisConfiguration
-   * @throws UncheckedIOException if the file is missing
-   */
-  public static ArtemisConfiguration fromFile(String path) {
-    Path configPath = Paths.get(path);
-    try {
-      return new ArtemisConfiguration(Configuration.fromToml(configPath, schema));
-    } catch (IOException e) {
-      throw new UncheckedIOException(e);
-    }
-  }
-
-  /**
-   * Reads configuration from a toml text.
-   *
-   * @param configText the toml text
-   * @return the new ArtemisConfiguration
-   */
-  public static ArtemisConfiguration fromString(String configText) {
-    return new ArtemisConfiguration(Configuration.fromToml(configText, schema));
+  // Network
+  private final String network;
+
+  // P2P
```

Comment:
(optional) At some point, it would be nice to start breaking up these sets of options into smaller config objects like `P2PConfig` etc.

Reply:
For follow-up PR.